### ISSUE TITLE
[UX] Respect the anonymous visitors setting

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3511]
**Steps To Reproduce**
1. Navigate to `/admin/config/people/roles`
2. Expand the "Anonymous visitors" fieldset, give a name to anonymous users other than the default "Anonymous" (I went with "Ghosts"), and finally save settings
3. Navigate to `/admin/config/people/permissions`
4. Navigate to `/admin/config/people/settings`

**Actual behavior**
- the column in `/admin/config/people/permissions` still says "ANONYMOUS" 👎 
- the radio labels in the "Who can register accounts?" setting say "Visitors" 👎

**Expected behavior**
- the column in `/admin/config/people/permissions` should have the same label as specified in step 2
- the radios in the "Who can register accounts?" setting should use the same label as specified in step 2

**Additional information**
Respective d.org issue: https://www.drupal.org/project/drupal/issues/749298

**Comment by klonos** [comment]
Turns out that making the header of the permissions table column respect the setting depends on #3510, because that value is pulled from the label of the role, rather than from the `'anonymous'` setting in `system.core.json`. So I just went with fixing only the radio labels in `/admin/config/people/settings`:

<img width="501" alt="screen shot 2019-01-31 at 4 03 36 am" src="https://user-images.githubusercontent.com/2423362/51998694-3b3a7480-250d-11e9-8fd5-ee60280d6d4c.png">

**Comment by klonos** [comment]
...I have noticed that the "Require email verification..." setting only applies for the 2nd and 3rd options of the "Who can register accounts" radio. I don't mean to scope-creep, but hiding it using `#states` would be a relatively small change. Should I add it to this PR, or separate issue?

**Comment by jenlampton** [comment]
I think that you may be misunderstanding the setting for `The name you would like used for anonymous visitors to your site`. It is not the same thing as the `Role name` which is what should be used in locations 6 and 7 as outlined in this issue. 

This setting controls what appears as the Username in the `Submitted by` area, when a node, or comment is displayed. 
<img width="1732" alt="screen shot 2019-01-31 at 12 16 15 pm" src="https://user-images.githubusercontent.com/397895/52082616-1ae0d900-2552-11e9-8ec8-48e9179f4e75.png">

<img width="1706" alt="screen shot 2019-01-31 at 12 17 20 pm" src="https://user-images.githubusercontent.com/397895/52082650-39df6b00-2552-11e9-9f76-c963b6bd0b8a.png">

I think this issue should work to rephrase the label for that field, so this is less confusing to others.

**Comment by BWPanda** [comment]
> approval is required by God

😅😂🤣
That should totally be the default name for the admin role!

**Comment by jenlampton** [comment]
We used to always name it "hal" back when I first started working at CH3.

**Comment by klonos** [comment]
> This setting controls what appears (to visitors) as the Username in the `Submitted by` area, when a node, or comment is displayed. 

Right. Sounds like that setting should not be in the "Roles" section 🤔 ...although I have no better place to suggest for it.

I have updated the PR to rename the fieldset (and the word "name"), simplify the field label, and add a description (based on @jenlampton's explanation):

![Screen Shot 2019-05-25 at 2 12 34 pm](https://user-images.githubusercontent.com/2423362/58364406-36079800-7ef7-11e9-92bf-dfe4f696f5ab.png)

**Comment by jenlampton** [comment]
Love the description text!
Maybe add "the" between "as" and "username" ?

**Comment by klonos** [comment]
...so I spent **a lot** of time trying to figure out how the whole "Who can register accounts" setting works, and it turns out that the confusion is due to a combination of assumptions, workarounds, and lack of comments in the code ...then trying to communicate all that in the UI 😞. Here's a new meta, with both a `TL;DR` section, and a "story of my life section": #3815.

I would especially love if anyone has some time to go through the 3 bullet points above the "Please let me know if I have any of the above wrong..." bolded bit, and verify my conclusions, or clarify where/if I got things wrong.

I will update the PR here, to only include the changes in the "Anonymous visitors name" fieldset...

**Comment by klonos** [comment]
...done.

**Comment by jenlampton** [comment]
There isn't a whole lot of difference between `The name you would like used for anonymous visitors to your site` and `The name that will be used for anonymous visitors`. I would love to see the label for this field rephrased to remove the confusion with role name, rather than adding `Not to be confused with the <em>Anonymous</em> user role` in the description.

How about something like `Anonymous author name to display for content and comments`. If we did that, we wouldn't really need any description text at all.

**Comment by klonos** [comment]
Sorry for the really long time to respond here. I'll update the PR and rebase, to get a sandbox going. Hopefully we can get some UX clarity about this in the upcoming 1.23.2 or at least 1.24.0 release.

**Comment by klonos** [comment]
This is what I currently have in the PR:
![image](https://user-images.githubusercontent.com/2423362/210458391-cd949cd5-c378-44c7-bb8c-2638cf5a8c6e.png)

So, in D7 and D8+ this setting lives in the "Account settings" form (`admin/config/people/settings`):
![image](https://user-images.githubusercontent.com/2423362/210458859-be97f24c-bb45-4f42-ad50-1c35f6a2c7ce.png)
![image](https://user-images.githubusercontent.com/2423362/210458962-18b4eafd-b5e8-4573-b2d5-a534e950fb58.png)

In D8+ the setting for which role is admin was moved to `admin/people/role-settings`:
![image](https://user-images.githubusercontent.com/2423362/210459131-f850f67a-a428-4b31-bd73-e80d6f798fd1.png)

Apparently, we moved both settings under `admin/config/people/roles`, but since the setting for anonymous vistors/authors is not role-related, I think that `admin/config/people/roles` is a bad place for it and it causes confusion. I think that it should be moved back to `admin/config/people/settings`.

@jenlampton? others?

**Comment by quicksketch** [comment]
How would you feel about moving the string for the Anonymous name to be under the configuration of the Anonymous role? That'd be stashing it in a pretty obscure spot, but also in a spot that makes sense (to me).

**Comment by klonos** [comment]
@quicksketch nope. ...as I mention in my previous comment, this setting is about a user (a username) - it has nothing to do with roles. Adding it under the configuration of the Anonymous role would make the confusion even worse.

So to sum up:
- one thing is a label for a user role
- the other thing is a username for user 0

**Comment by kiamlaluno** [comment]
I would put that settings in admin/config/people/settings.

**Comment by olafgrabienski** [comment]
In my opinion, `admin/config/people/roles` isn't ideal, but it's not such a bad place, so I'd like to focus first on the text changes (AKA baby steps) and consider separately if we need a better place for the setting. Re text changes, the current PR is already much better than before.

@klonos Are you also planning to integrate Jen's feedback from https://github.com/backdrop/backdrop-issues/issues/3511#issuecomment-574895302, or not.

**Comment by klonos** [comment]
> @klonos Are you also planning to integrate Jen's feedback from https://github.com/backdrop/backdrop-issues/issues/3511#issuecomment-574895302, or rather not?

Not sure if any of that is still relevant, but I'll let @jenlampton re-review and reconsider, as I've made some additional changes to the UI text since.

Trying to break down Jen's feedback from back in 2020...

> There isn't a whole lot of difference between `The name you would like used for anonymous visitors to your site` and `The name that will be used for anonymous visitors`. 

This text has now been simplified to `Name to be used for anonymous authors`. There is a `:` at the end of that text in the PR, but since we are not including semicolons in field labels, I would like to get rid of it. Will wait for further feed back on the label text before I do that.

> I would love to see the label for this field rephrased to remove the confusion with role name, rather than adding `Not to be confused with the <em>Anonymous</em> user role` in the description.

I have done that I believe, but perhaps we could change the word "Name" with "Username" in the label, in order to distinguish things even further. Fresh feedback on that please?

> How about something like `Author name to display for Anonymous visitors on content and comments`. If we did that, we wouldn't really need any description text at all.

That is really lengthy for a label. It feels like placing text that belongs in help text on the label, which defeats the purpose of the label (labels are to be a short as possible - any clarification goes in the help text).

Lastly, I don't see anything wrong with the "Not to be confused with the %anonymous user role." bit of the help text. It is precise and to the point - exactly what we want people to avoid confusing, which is what it seems to be happening. Even chronic Drupal/Backdrop users seem to be tripped by this all the time ...even yours truly (see the issue summary to see what I mean - I was the one to suggest that the heading in the permissions page should change to reflect this setting to begin with).
